File: src/WebApi/CompanyXApi/CompanyXApi/ViewModels/PaginatedResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CompanyXApi.ViewModels
{
    public class PaginatedResponse<TEntity> where TEntity : class
    {
        public int PageIndex { get; private set; }
        public int Count { get; private set; }
        public long Total { get; private set; }
        public IEnumerable<TEntity> Employees { get; private set; }

        public PaginatedResponse(int pageIndex, int count, long total, IEnumerable<TEntity> employees)
        {
            this.PageIndex = pageIndex;
            this.Count = count;
            this.Total = total;
            this.Employees = employees;
        }
    }
}

File: src/WebApp/CompanyXWeb/ClientApp/src/app/Models/PaginatedResponse.ts
import { DisplayEmployeeVM } from './DisplayEmployeeVM'
export class PaginatedResponse {
  public readonly pageIndex: number;
  public readonly count: number;
  public readonly total: number;
  public readonly employees: DisplayEmployeeVM[];


  constructor(pageIndex: number,count: number, total: number, employees: DisplayEmployeeVM[]) {
    this.pageIndex = pageIndex;
    this.count = count;
    this.total = total;
    this.employees = employees;
  }

}
